src/app-layout/utils/use-drawer-focus-control.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { DependencyList, RefObject, useEffect, useRef } from 'react';
+import { ButtonProps } from '../../button/interfaces';
+
+export type DrawerLastInteraction = { type: 'open' } | { type: 'close' };
+
+export interface DrawerFocusControlRefs {
+  toggle: RefObject<ButtonProps.Ref>;
+  slider: RefObject<HTMLDivElement>;
+}
+interface DrawerFocusControlState {
+  refs: DrawerFocusControlRefs;
+  setLastInteraction: (interaction: DrawerLastInteraction) => void;
+}
+
+export function useDrawerFocusControl(dependencies: DependencyList): DrawerFocusControlState {
```

Comment:
The long term plan was to consolidate focus control in a single controller. 

So instead of creating a new controller for the drawers, we need to refactor existing ones to support an array of possible focusables

Reply:
Note from live discussion: I had to add this (separate from the VR version) because of the focus state on the resize handle. We will consolidate after adding resize to VR drawers.